File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-6.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int res = nums[0];
        int lo = 0;
        int hi = nums.size() - 1;
        int mid = 0;
        while (lo <= hi) {
            if (nums[lo] < nums[hi]) {
                res = min(res, nums[lo]);
                break;
            }

            mid = (lo + hi) / 2;
            res = min(res, nums[mid]);
            if (nums[mid] >= nums[lo]) {
                lo = mid + 1;
            } else {
                hi = mid - 1;
            }
        }
        return res;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-6.py
import heapq

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0

        map = {}
        minVal = min(nums)
        print("min = " + str(minVal))
        maxVal = max(nums)
        print("max = " + str(maxVal))

        for num in nums:
            map[num] = True

        currLen = 1
        maxLen = 1
        
        for i in range(minVal + 1, maxVal + 1):
            if i in map:
                currLen += 1
            else:
                currLen = 0
            maxLen = max(currLen, maxLen)

        return maxLen